microservices/workers/src/Worker/Rates.php

```diff
@@ -98,86 +77,156 @@ public function import(GearmanJob $serializedJob)
+                    $rates[] = sprintf('("%s", %d)',
+                        $rateName,
+                        $brandId
+                    );
+
+                    $tpRates[] = sprintf('("%s", "%s", "%ss", %s, %s)',
+                        $line["Connection charge"],
+                        $line["Per minute charge"],
+                        $line["Charge period"],
+                        sprintf('(SELECT tag FROM Rates WHERE name = "%s" LIMIT 1)', $rateName),
+                        sprintf('(SELECT id FROM Rates WHERE name = "%s" LIMIT 1)', $rateName)
+                    );
+
+                    $ratesCache[$rateName] = true;
+                }
+            }
+
+        }
+
+        try {
+            $this->em->getConnection()->beginTransaction();
+
+            if (!empty($destinations)) {
+                $this->em->getConnection()->executeQuery(
+                    'INSERT IGNORE INTO Destinations (name_en, name_es, description_en, description_es, brandId) VALUES ' . implode(",", $destinations)
+                );
+
+                $this->em->getConnection()->executeQuery(
+                    'UPDATE Destinations SET tag = CONCAT("b", brandId, "dst", id)'
+                );
+            }
+
+            if (!empty($tpDestinations)) {
+                $this->em->getConnection()->executeQuery(
+                    'INSERT IGNORE INTO tp_destinations (prefix, tag, destinationId) VALUES ' . implode(",", $tpDestinations)
+                );
+            }
+
+            if (!empty($rates)) {
+                $this->em->getConnection()->executeQuery(
+                    'INSERT IGNORE INTO Rates (name, brandId) VALUES ' . implode(",", $rates)
+                );
+
+
+                $this->em->getConnection()->executeQuery(
+                    'UPDATE IGNORE Rates SET tag = CONCAT("b", brandId, "rt", id)'
+                );
+            }
+
+            if (!empty($tpRates)) {
+                $this->em->getConnection()->executeQuery(
+                    'INSERT IGNORE INTO tp_rates (connect_fee, rate, rate_increment, tag, rateId ) VALUES ' . implode(",", $tpRates)
+                );
+            }
+
+            $this->em->getConnection()->executeQuery(
+                "INSERT INTO DestinationRates (name_en, name_es, description_es, description_en, brandId) VALUES ('$destinationRateName', '$destinationRateName', 'Imported From CSV', 'Imported From CSV', $brandId)"
+            );
+
+            $this->em->getConnection()->executeQuery(
+                'UPDATE DestinationRates SET tag = CONCAT("b", brandId, "dr", id)'
+            );
+
+            $this->em->getConnection()->executeQuery(
+                'INSERT INTO tp_destination_rates (destinations_tag, destinationId, rates_tag, rateId, tag, destinationRateId) VALUES ' . implode(",", $tpDestinationRates)
+            );
+
+
+            $this->em->getConnection()->commit();
+        } catch (DBALException $exception) {
+            echo $exception->getMessage();
```

Comment:
You should run $this->em->close() and  $this->em->getConnection()->rollback() here